### PR TITLE
Dev Branch -> Master Merge

### PR DESCRIPTION
**Pull request by StephenHodgson**
Merges all the changes from the dev branch into master.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,34 +10,78 @@ MixedRealityToolkit-Unity uses code from the base [MixedRealityToolkit](https://
...
+| ![Spatial Sound](External/ReadMeImages/MRTK170802_Short_09.png) [Spatial Sound](Assets/HoloToolkit/SpatialSound/README.md)                        | ![UX Controls](External/ReadMeImages/MRTK170802_Short_10.png) [UX Controls](Assets/HoloToolkit/UX/README.md)| ![Utilities](External/ReadMeImages/MRTK170802_Short_11.png) [Utilities](Assets/HoloToolkit/Utilities/README.md)                 | 
+| Scripts to help plug spatial audio into your application.                                                                                         | Building blocks for creating good UX in your application like common controls.                              | Common helpers and tools that you can leverage in your application.                                                             |
+| ![Spatial Understanding](External/ReadMeImages/MRTK170802_Short_06.png) [Spatial Understanding](Assets/HoloToolkit/SpatialUnderstanding/README.md)| ![Build](External/ReadMeImages/MRTK170802_Short_12.png) [Build](Assets/HoloToolkit/BuildAndDeploy/README.md)| ![Boundary](External/ReadMeImages/MRTK170802_Short_06.png) [Build](Assets/HoloToolkit/Boundary/README.md)                       |
```
Comment (keveleigh) [inline README.md]: I think the final `Build` link is supposed to be `Boundary`.

File: Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace HoloToolkit.Unity
     public class ProjectSettingsWindow : AutoConfigureWindow<ProjectSettingsWindow.ProjectSetting>
     {
         private const string SharingServiceURL = "https://raw.githubusercontent.com/Microsoft/MixedRealityToolkit-Unity/master/External/HoloToolkit/Sharing/Server/SharingService.exe";
-        private const string InputManagerAssetURL = "https://raw.githubusercontent.com/Microsoft/MixedRealityToolkit-Unity/master/ProjectSettings/InputManager.asset";
+        private const string InputManagerAssetURL = "https://raw.githubusercontent.com/Microsoft/MixedRealityToolkit-Unity/Dev_Unity_2017.2.0/ProjectSettings/InputManager.asset";
```
Comment (keveleigh) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs]: Should this change to master now? Pending my PR though, I guess.

File: Assets/HoloToolkit-Examples/Utilities/Scripts/TextToSpeechOnTapTest.cs [hunk 3]
```diff
@@ -5,9 +5,6 @@
 
 namespace HoloToolkit.Unity.InputModule.Tests
 {
-    /// <summary>
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Utilities/Scripts/TextToSpeechOnTapTest.cs]: Looks like some of the changes in this file were lost.

File: Assets/HoloToolkit-Examples/Utilities/Scripts/TextToSpeechOnTapTest.cs [hunk 4]
```diff
@@ -16,16 +13,23 @@ private void Awake()
         {
             textToSpeech = GetComponent<TextToSpeech>();
         }
-
+
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Utilities/Scripts/TextToSpeechOnTapTest.cs]: nit: rogue spaces

File: Assets/HoloToolkit-Examples/SpatialMapping/Scenes/PlaneFinding.unity [hunk 5]
```diff
@@ -38,11 +38,19 @@ RenderSettings:
   m_ReflectionIntensity: 1
   m_CustomReflection: {fileID: 0}
   m_Sun: {fileID: 0}
+<<<<<<< HEAD
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/SpatialMapping/Scenes/PlaneFinding.unity]: Looks like some unfinished merge conflicts made their way here.

File: Assets/HoloToolkit-Examples/Input/Scripts/TapResponder.cs [hunk 6]
```diff
@@ -11,10 +12,12 @@ namespace HoloToolkit.Unity.InputModule.Tests
     /// </summary>
     public class TapResponder : MonoBehaviour, IInputClickHandler
     {
-        public void OnInputClicked(InputClickedEventData eventData)
+         public void OnInputClicked(InputClickedEventData eventData)
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Input/Scripts/TapResponder.cs]: nit: extra space

File: Assets/HoloToolkit-Examples/Input/Scripts/InputTest.cs [hunk 7]
```diff
@@ -42,36 +45,43 @@ public void OnFocusExit()
...
             Debug.LogFormat("OnHoldCanceled\r\nSource: {0}  SourceId: {1}", eventData.InputSource, eventData.SourceId);
+            eventData.Use(); // Mark the event as used, so it doesn't fall through to other handlers.
         }
 
         public void OnManipulationStarted(ManipulationEventData eventData)
         {
-            Debug.LogFormat("OnManipulationStarted\r\nSource: {0}  SourceId: {1}\r\nCumulativeDelta: {2} {3} {4}",
-                eventData.InputSource,
+            Debug.LogFormat("OnManipulationStarted\r\nSource: {0}  SourceId: {1}\r\nCumulativeDelta: {2} {3} {4}", 
+                eventData.InputSource,
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/Input/Scripts/InputTest.cs]: nit: extra space on the above two lines

File: Assets/HoloToolkit-Examples/ColorPicker/GazeableColorPicker.cs [hunk 8]
```diff
@@ -27,6 +27,10 @@ private void Update()
 
         private void UpdatePickedColor(PickedColorCallback cb)
         {
+            GameObject hitObject = GazeManager.Instance.HitObject;
```
Comment (keveleigh) [inline Assets/HoloToolkit-Examples/ColorPicker/GazeableColorPicker.cs]: This looks like duplicate code compared to the existing lines below.

**Comment by keveleigh** [comment]
Also (I wasn't able to comment during the review because there weren't any changes, but I'm seeing a 
`...hUNET/Scenes/SharingWithUnetExample.unity → ...ut/Scenes/InputNavigationRotateTest.unity` change that claims to have no changes. I think that's accurate, because opening `InputNavigationRotateTest.unity` in the current dev branch looks to be the SharingWithUnetExample (though it's missing a prefab). Are you seeing the same thing?

File: Assets/HoloToolkit-Examples/SpatialMapping/Scenes/PlaneFinding.unity (hunk 5 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit-Examples/SpatialMapping/Scenes/PlaneFinding.unity]: good catch!

File: Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs (hunk 2 above)
Comment (StephenHodgson) [reply Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs]: Yup

**Comment by StephenHodgson** [comment]
So the SharingWithUnet used to live in the examples folder, but in the dev branch it's in the main toolklit.

Should we move it back to examples?
A lot of the assets in the folder are example type assets.

**Comment by StephenHodgson** [comment]
> InputNavigationRotateTest.unity

I couldn't find any assets with the GUID that matches the prefab that was removed.

**Comment by StephenHodgson** [comment]
> some of the scripts would be useful to keep in the actual Toolkit folder

I agree.

We can work on that later.

**Comment by keveleigh** [comment]
> It's fixed

Awesome! Any chance you found the real `InputNavigationRotateTest.unity` as well?

**Comment by StephenHodgson** [comment]
Indeed. I just took the changes from the Dev branch.

**Comment by StephenHodgson** [comment]
Just want to do one pass at the Motion Controller example we merged in and wrap it, and merge the latest dev into this branch.

Sound good?

**Comment by StephenHodgson** [comment]
@keveleigh, okay that's everything.  I merged in all the new stuff from the Dev branch with the motion controllers

**Comment by StephenHodgson** [comment]
Okay _now_ it's ready. Didn't build at first.

**Comment by keveleigh** [comment]
image
Did this get in another way or become unneeded?

**Comment by StephenHodgson** [comment]
> Did this get in another way or become unneeded?

It was already fixed.  I just double checked the meta file, and removed the missing prefab reference.

**Comment by StephenHodgson** [comment]
Whooo We've got our releases boys!

Gonna go work on that now.

**Pull request merged**